src/databricks/labs/ucx/source_code/notebooks/cells.py

```diff
@@ -496,27 +494,57 @@ def _make_commands_for_magic_command_call_nodes(cls, nodes: list[Call]):
         for node in nodes:
             arg = node.args[0]
             if isinstance(arg, Const):
-                yield MagicCommand(node, arg.value)
+                yield MagicLine(node, arg.value)
 
     def __init__(self, node: NodeNG, command: bytes):
         super().__init__(node)
         self._command = command.decode()
 
-    def build_dependency_graph(self, graph: DependencyGraph) -> list[DependencyProblem]:
+    def as_magic(self) -> MagicCommand | None:
         if self._command.startswith("%pip") or self._command.startswith("!pip"):
-            cmd = PipMagic(self._command)
-            return cmd.build_dependency_graph(graph)
+            return PipCommand(self._command)
+        if self._command.startswith("%run"):
+            return RunCommand(self._command)
+        return None
+
+    def build_dependency_graph(self, graph: DependencyGraph) -> list[DependencyProblem]:
+        magic = self.as_magic()
+        if magic is not None:
+            return magic.build_dependency_graph(graph)
         problem = DependencyProblem.from_node(
             code='unsupported-magic-line', message=f"magic line '{self._command}' is not supported yet", node=self.node
         )
         return [problem]
 
 
-class PipMagic:
+class MagicCommand(ABC):
 
     def __init__(self, code: str):
         self._code = code
 
+    @abstractmethod
+    def build_dependency_graph(self, graph: DependencyGraph) -> list[DependencyProblem]: ...
+
+
+class RunCommand(MagicCommand):
+
+    def build_dependency_graph(self, graph: DependencyGraph) -> list[DependencyProblem]:
+        # TODO
```

Comment:
implement this

Reply:
I was planning to do this in a separate PR since it's unrelated, see https://github.com/databrickslabs/ucx/issues/2155

Reply:
it hides a bug of mishandling `%run` magic and command. it's almost `return graph.register_notebook(self.notebook_path)` + error checking. and one more functional test. Implement it in this PR, otherwise it'll be forgotten. you can make a PR to this PR.

Reply:
I agree. I'll prioritize #2155

Reply:
It's done in PR #2164